src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,269 @@
+        past_feat_dynamic_real=past_feat_dynamic_real,
+        prediction_length=prediction_length,
+    )
+
+
+def dataframeTS_to_dataentry(
+    data: pd.DataFrame,
+    target: Union[str, List[str]],
+    timestamp: str,
+    feat_dynamic_real: List[str] = [],
+    feat_dynamic_cat: List[str] = [],
+    feat_static_real: List[str] = [],
+    feat_static_cat: List[str] = [],
+    past_feat_dynamic_real: List[str] = [],
+) -> DataEntry:
+    """
+    Converts a single time series (uni- or multi-variate) that is given in
+    a pd.DataFrame format to a DataEntry.
+
+    Parameters
+    ----------
+    data: pd.DataFrame
+        pd.DataFrame containing at least timestamp, target and item_id columns.
+    target: str or List[str]
+        Name of the column that contains the target time series.
+        For multivariate targets `target` is expecting a list of column names.
+    timestamp: str
+        Name of the column that contains the timestamp information.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features only for the history.
+
+    Returns
+    -------
+    DataEntry
+        A dictionary with at least `target` and `start` field.
+    """
+    dataentry = {
+        FieldName.START: data.loc[:, timestamp].iloc[0],
+    }
+
+    def set_field(fieldname, col_names, f=lambda x: x):
+        if col_names:
+            dataentry[fieldname] = [
+                f(data.loc[:, n].to_list()) for n in col_names
+            ]
+
+    if isinstance(target, str):
+        dataentry[FieldName.TARGET] = data.loc[:, target].to_list()
+    else:
+        set_field(FieldName.TARGET, target)
+    set_field(FieldName.FEAT_DYNAMIC_REAL, feat_dynamic_real)
+    set_field(FieldName.FEAT_DYNAMIC_CAT, feat_dynamic_cat)
+    set_field(FieldName.FEAT_STATIC_REAL, feat_static_real, lambda x: x[0])
+    set_field(FieldName.FEAT_STATIC_CAT, feat_static_cat, lambda x: x[0])
+    set_field(FieldName.PAST_FEAT_DYNAMIC_REAL, past_feat_dynamic_real)
+    return dataentry
+
+
+def prepare_prediction_data(
+    dataentry: DataEntry, prediction_length: int
+) -> DataEntry:
+    """
+    Removes `prediction_length` values from `target` and `past_feat_dynamic_real`.
+    Works in univariate and multivariate case.
+
+    >>> prepare_prediction_data({"target": np.array([1., 2., 3., 4.])}, prediction_length=2)
+    {'target': array([1., 2.])}
+    """
+    for fname in [FieldName.TARGET, FieldName.PAST_FEAT_DYNAMIC_REAL]:
+        if fname in dataentry:
+            dataentry[fname] = dataentry[fname][..., :-prediction_length]
```

Comment:
This looks like it's doing it in-place. What happens when you iterate twice over the dataset?